<!DOCTYPE html>
<html lang="zh-cn">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
    <title>API Test Report</title>
    <meta name="generator" content="BSTestRunner 0.8.4"/>
    <link rel="stylesheet" href="http://cdn.bootcss.com/bootstrap/3.3.0/css/bootstrap.min.css">
    
<style type="text/css" media="screen">

/* -- css div popup ------------------------------------------------------------------------ */
.popup_window {
    display: none;
    position: relative;
    left: 0px;
    top: 0px;
    /*border: solid #627173 1px; */
    padding: 10px;
    background-color: #99CCFF;
    font-family: "Lucida Console", "Courier New", Courier, monospace;
    text-align: left;
    font-size: 10pt;
    width: 500px;
}

/* -- report ------------------------------------------------------------------------ */

#show_detail_line .label {
    font-size: 85%;
    cursor: pointer;
}

#show_detail_line {
    margin: 2em auto 1em auto;
}

#total_row  { font-weight: bold; }
.hiddenRow  { display: none; }
.testcase   { margin-left: 2em; }

</style>


    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
    <!--[if lt IE 9]>
      <script src="http://cdn.bootcss.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="http://cdn.bootcss.com/respond.js/1.4.2/respond.min.js"></script>
    <![endif]-->
  </head>
<body>
<script language="javascript" type="text/javascript"><!--
output_list = Array();

/* level - 0:Summary; 1:Failed; 2:All */
function showCase(level) {
    trs = document.getElementsByTagName("tr");
    for (var i = 0; i < trs.length; i++) {
        tr = trs[i];
        id = tr.id;
        if (id.substr(0,2) == 'ft') {
            if (level < 1) {
                tr.className = 'hiddenRow';
            }
            else {
                tr.className = '';
            }
        }
        if (id.substr(0,2) == 'pt') {
            if (level > 1) {
                tr.className = '';
            }
            else {
                tr.className = 'hiddenRow';
            }
        }
    }
}


function showClassDetail(cid, count) {
    var id_list = Array(count);
    var toHide = 1;
    for (var i = 0; i < count; i++) {
        tid0 = 't' + cid.substr(1) + '.' + (i+1);
        tid = 'f' + tid0;
        tr = document.getElementById(tid);
        if (!tr) {
            tid = 'p' + tid0;
            tr = document.getElementById(tid);
        }
        id_list[i] = tid;
        if (tr.className) {
            toHide = 0;
        }
    }
    for (var i = 0; i < count; i++) {
        tid = id_list[i];
        if (toHide) {
            document.getElementById('div_'+tid).style.display = 'none'
            document.getElementById(tid).className = 'hiddenRow';
        }
        else {
            document.getElementById(tid).className = '';
        }
    }
}


function showTestDetail(div_id){
    var details_div = document.getElementById(div_id)
    var displayState = details_div.style.display
    // alert(displayState)
    if (displayState != 'block' ) {
        displayState = 'block'
        details_div.style.display = 'block'
    }
    else {
        details_div.style.display = 'none'
    }
}


function html_escape(s) {
    s = s.replace(/&/g,'&amp;');
    s = s.replace(/</g,'&lt;');
    s = s.replace(/>/g,'&gt;');
    return s;
}

/* obsoleted by detail in <div>
function showOutput(id, name) {
    var w = window.open("", //url
                    name,
                    "resizable,scrollbars,status,width=800,height=450");
    d = w.document;
    d.write("<pre>");
    d.write(html_escape(output_list[id]));
    d.write("\n");
    d.write("<a href='javascript:window.close()'>close</a>\n");
    d.write("</pre>\n");
    d.close();
}
*/
--></script>

<div class="container">
    <div class='heading'>
<h1>API Test Report</h1>
<p><strong>Start Time:</strong> 2018-10-02 09:21:03</p>
<p><strong>Duration:</strong> 0:00:03.246701</p>
<p><strong>Status:</strong> <span class="text text-success">Pass <strong>5</strong></span> <span class="text text-danger">Failure <strong>1</strong></span> <span class="text text-warning">Error <strong>4</strong></span></p>

<p class='description'>Django Restful API Test Report</p>
</div>


    
<p id='show_detail_line'>
<span class="label label-primary" onclick="showCase(0)">Summary</span>
<span class="label label-danger" onclick="showCase(1)">Failed</span>
<span class="label label-default" onclick="showCase(2)">All</span>
</p>
<table id='result_table' class="table">
    <thead>
        <tr id='header_row'>
            <th>Test Group/Test case</td>
            <th>Count</td>
            <th>Pass</td>
            <th>Fail</td>
            <th>Error</td>
            <th>View</td>
        </tr>
    </thead>
    <tbody>
        
<tr class='text text-warning'>
    <td>test_django_restful.GroupTest</td>
    <td>4</td>
    <td>2</td>
    <td>0</td>
    <td>2</td>
    <td><a class="btn btn-xs btn-primary"href="javascript:showClassDetail('c1',4)">Detail</a></td>
</tr>

<tr id='ft1.1' class='text text-success'>
    <td class='text text-warning'><div class='testcase'>test_001_group_developer1</div></td>
    <td colspan='5' align='center'>

    <!--css div popup start-->
    <a class="popup_link btn btn-xs btn-default" onfocus='this.blur();' href="javascript:showTestDetail('div_ft1.1')" >
        error</a>

    <div id='div_ft1.1' class="popup_window">
        <div style='text-align: right;cursor:pointer'>
        <a onfocus='this.blur();' onclick="document.getElementById('div_ft1.1').style.display = 'none' " >
           [x]</a>
        </div>
        <pre>
        
ft1.1: Traceback (most recent call last):
  File "D:\django_restful\api\test_project\test_django_restful.py", line 89, in test_001_group_developer1
    self.assertEqual(result['name'],'Developer')
KeyError: 'name'


        </pre>
    </div>
    <!--css div popup end-->

    </td>
</tr>

<tr id='pt1.2' class='hiddenRow'>
    <td class='text text-success'><div class='testcase'>test_002_add_group2</div></td>
    <td colspan='5' align='center'>pass</td>
</tr>

<tr id='ft1.3' class='text text-success'>
    <td class='text text-warning'><div class='testcase'>test_003_update_group3</div></td>
    <td colspan='5' align='center'>

    <!--css div popup start-->
    <a class="popup_link btn btn-xs btn-default" onfocus='this.blur();' href="javascript:showTestDetail('div_ft1.3')" >
        error</a>

    <div id='div_ft1.3' class="popup_window">
        <div style='text-align: right;cursor:pointer'>
        <a onfocus='this.blur();' onclick="document.getElementById('div_ft1.3').style.display = 'none' " >
           [x]</a>
        </div>
        <pre>
        
ft1.3: Traceback (most recent call last):
  File "D:\django_restful\api\test_project\test_django_restful.py", line 111, in test_003_update_group3
    self.assertEqual(result['name'],'Boss')
KeyError: 'name'


        </pre>
    </div>
    <!--css div popup end-->

    </td>
</tr>

<tr id='pt1.4' class='hiddenRow'>
    <td class='text text-success'><div class='testcase'>test_004_delete_group4</div></td>
    <td colspan='5' align='center'>pass</td>
</tr>

<tr class='text text-warning'>
    <td>test_django_restful.UserTest</td>
    <td>6</td>
    <td>3</td>
    <td>1</td>
    <td>2</td>
    <td><a class="btn btn-xs btn-primary"href="javascript:showClassDetail('c2',6)">Detail</a></td>
</tr>

<tr id='ft2.1' class='text text-success'>
    <td class='text text-warning'><div class='testcase'>test_001_no_auth2: 没有登陆的查询的用例</div></td>
    <td colspan='5' align='center'>

    <!--css div popup start-->
    <a class="popup_link btn btn-xs btn-default" onfocus='this.blur();' href="javascript:showTestDetail('div_ft2.1')" >
        error</a>

    <div id='div_ft2.1' class="popup_window">
        <div style='text-align: right;cursor:pointer'>
        <a onfocus='this.blur();' onclick="document.getElementById('div_ft2.1').style.display = 'none' " >
           [x]</a>
        </div>
        <pre>
        
ft2.1: Traceback (most recent call last):
  File "C:\python\lib\site-packages\urllib3\connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "C:\python\lib\site-packages\urllib3\connectionpool.py", line 384, in _make_request
    six.raise_from(e, None)
  File "&lt;string&gt;", line 2, in raise_from
  File "C:\python\lib\site-packages\urllib3\connectionpool.py", line 380, in _make_request
    httplib_response = conn.getresponse()
  File "C:\python\lib\http\client.py", line 1198, in getresponse
    response.begin()
  File "C:\python\lib\http\client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "C:\python\lib\http\client.py", line 266, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\python\lib\site-packages\requests\adapters.py", line 445, in send
    timeout=timeout
  File "C:\python\lib\site-packages\urllib3\connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "C:\python\lib\site-packages\urllib3\util\retry.py", line 367, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "C:\python\lib\site-packages\urllib3\packages\six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "C:\python\lib\site-packages\urllib3\connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "C:\python\lib\site-packages\urllib3\connectionpool.py", line 384, in _make_request
    six.raise_from(e, None)
  File "&lt;string&gt;", line 2, in raise_from
  File "C:\python\lib\site-packages\urllib3\connectionpool.py", line 380, in _make_request
    httplib_response = conn.getresponse()
  File "C:\python\lib\http\client.py", line 1198, in getresponse
    response.begin()
  File "C:\python\lib\http\client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "C:\python\lib\http\client.py", line 266, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\django_restful\api\test_project\ff_fz.py", line 7, in get
    self.r = requests.get(base_url)
  File "C:\python\lib\site-packages\requests\api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "C:\python\lib\site-packages\requests\api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "C:\python\lib\site-packages\requests\sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\python\lib\site-packages\requests\sessions.py", line 644, in send
    history = [resp for resp in gen] if allow_redirects else []
  File "C:\python\lib\site-packages\requests\sessions.py", line 644, in &lt;listcomp&gt;
    history = [resp for resp in gen] if allow_redirects else []
  File "C:\python\lib\site-packages\requests\sessions.py", line 222, in resolve_redirects
    **adapter_kwargs
  File "C:\python\lib\site-packages\requests\sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "C:\python\lib\site-packages\requests\adapters.py", line 495, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\django_restful\api\test_project\test_django_restful.py", line 23, in test_001_no_auth2
    r=self.b.get(self.base_url)
  File "D:\django_restful\api\test_project\ff_fz.py", line 9, in get
    print(u'错误原因为: %s' % str(e))
  File "C:\python\lib\BSTestRunner.py", line 131, in write
    self.fp.write(to_unicode(s))
  File "C:\python\lib\BSTestRunner.py", line 120, in to_unicode
    return unicode(s)
NameError: name 'unicode' is not defined


        </pre>
    </div>
    <!--css div popup end-->

    </td>
</tr>

<tr id='pt2.2' class='hiddenRow'>
    <td class='text text-success'><div class='testcase'>test_002_add_user3: 添加数据用例</div></td>
    <td colspan='5' align='center'>pass</td>
</tr>

<tr id='pt2.3' class='hiddenRow'>
    <td class='text text-success'><div class='testcase'>test_003_update_user4: 更新email用例</div></td>
    <td colspan='5' align='center'>pass</td>
</tr>

<tr id='ft2.4' class='text text-success'>
    <td class='text text-danger'><div class='testcase'>test_004_delete_user5: 删除数据的用例</div></td>
    <td colspan='5' align='center'>

    <!--css div popup start-->
    <a class="popup_link btn btn-xs btn-default" onfocus='this.blur();' href="javascript:showTestDetail('div_ft2.4')" >
        fail</a>

    <div id='div_ft2.4' class="popup_window">
        <div style='text-align: right;cursor:pointer'>
        <a onfocus='this.blur();' onclick="document.getElementById('div_ft2.4').style.display = 'none' " >
           [x]</a>
        </div>
        <pre>
        
ft2.4: Traceback (most recent call last):
  File "D:\django_restful\api\test_project\test_django_restful.py", line 60, in test_004_delete_user5
    self.assertEqual(r.status_code,204)
AssertionError: 404 != 204


        </pre>
    </div>
    <!--css div popup end-->

    </td>
</tr>

<tr id='pt2.5' class='hiddenRow'>
    <td class='text text-success'><div class='testcase'>test_005_get_user6: 查看数据的用例</div></td>
    <td colspan='5' align='center'>pass</td>
</tr>

<tr id='ft2.6' class='text text-success'>
    <td class='text text-warning'><div class='testcase'>test_1: 数据初始化</div></td>
    <td colspan='5' align='center'>

    <!--css div popup start-->
    <a class="popup_link btn btn-xs btn-default" onfocus='this.blur();' href="javascript:showTestDetail('div_ft2.6')" >
        error</a>

    <div id='div_ft2.6' class="popup_window">
        <div style='text-align: right;cursor:pointer'>
        <a onfocus='this.blur();' onclick="document.getElementById('div_ft2.6').style.display = 'none' " >
           [x]</a>
        </div>
        <pre>
        
ft2.6: Traceback (most recent call last):
  File "D:\django_restful\api\test_project\test_django_restful.py", line 17, in test_1
    self.db.test_cz()
  File "D:\django_restful\api\test_project\mysql_action.py", line 57, in test_cz
    db.init_data(datas)
  File "D:\django_restful\api\test_project\mysql_action.py", line 49, in init_data
    print(d)
  File "C:\python\lib\BSTestRunner.py", line 131, in write
    self.fp.write(to_unicode(s))
  File "C:\python\lib\BSTestRunner.py", line 120, in to_unicode
    return unicode(s)
NameError: name 'unicode' is not defined


        </pre>
    </div>
    <!--css div popup end-->

    </td>
</tr>

    </tbody>
    <tfoot>
        <tr id='total_row'>
            <td>Total</td>
            <td>10</td>
            <td class="text text-success">5</td>
            <td class="text text-danger">1</td>
            <td class="text text-warning">4</td>
            <td>&nbsp;</td>
        </tr>
    </tfoot>
</table>

    <div id='ending'>&nbsp;</div>
</div>

</body>
</html>
